src/module.c

```diff
@@ -3644,12 +3645,14 @@ int VM_ReplyWithLongDouble(ValkeyModuleCtx *ctx, long double ld) {
  * The command returns VALKEYMODULE_ERR if the format specifiers are invalid
  * or the command name does not belong to a known command. */
 int VM_Replicate(ValkeyModuleCtx *ctx, const char *cmdname, const char *fmt, ...) {
-    struct serverCommand *cmd;
+    struct serverCommand *cmd = NULL;
     robj **argv = NULL;
     int argc = 0, flags = 0, j;
     va_list ap;
+    int slot = -1;
 
-    if (!ctx->module || !(ctx->module->options & VALKEYMODULE_OPTIONS_SKIP_COMMAND_VALIDATION)) {
+    if ((server.cluster_enabled && clusterIsAnySlotExporting()) || !ctx->module ||
```

Comment:
I don't think we should error only when slot migration is ongoing. Something might work perfect and then break at some random point in the future with slot migration. With that said, always adding this becomes a breaking change. Neither option feels great.

Reply:
Today it only runs when a slot export is ongoing. So an application might be working, then break only when atomic slot migration happens. Ideally the behavior shouldn't change when slot migration is running, either from performance or errors.

Reply:
I guess honestly I would rather abort the slot migration than error.